lib/pbench/test/unit/server/test_user_management_cli.py

```diff
@@ -133,20 +134,21 @@ def test_user_creation_with_invalid_role(client, server_config, pytestconfig):
             )
             assert result.exit_code == 2
             assert (
-                result.stderr_bytes
-                == b"Usage: user-create [OPTIONS]\nTry 'user-create --help' for help.\n"
-                + b"\nError: Invalid value for '--role': invalid choice: ADMN. (choose from ADMIN)\n"
+                result.stderr.find(
+                    "Invalid value for '--role': invalid choice: ADMN. (choose from ADMIN)"
+                )
+                > -1
             )
 
     @staticmethod
     @responses.activate
     def test_valid_user_delete(client, server_config, pytestconfig):
         with client:
             runner = CliRunner(mix_stderr=False)
-            result = TestUserManagement.register_user(server_config, pytestconfig)
-            assert result.exit_code == 0, result.stderr
+            result = TestUserManagement.register_user()
 
             result = runner.invoke(user_delete, args=[TestUserManagement.USER_TEXT])
+            assert result.stdout == "User test_user deleted\n"
             assert result.exit_code == 0, result.stderr
             assert result.stdout == "User test_user deleted\n"
```

Comment:
It looks like you have duplicate assertions on lines 151 and 153.

Should this message be sent to `stdout` or `stderr`?  I'll grant that it's a "success message" and not an "error message", but it is an indication of the operation of the command, not something that the command is supposed to produce.

Reply:
Shouldn't this be a stdout, this is not an error, it says user is deleted successfully which is what the command is supposed to do, in case the user is not deleted it's an error right?

Reply:
> @webbnh Shouldn't this be a stdout, this is not an error, it says user is deleted successfully which is what the command is supposed to do, in case the user is not deleted it's an error right?

`stderr` is named for "error output" mostly because the ancient UNIX CLI standard is designed to support piping and filtering commands. The only "out of band" messages (that is, output aside from what the user actually requested) should be errors, and they should not be in the standard output (`stdout`) that would be consumed by a pipeline stage. Out of band messages like "success" shouldn't exist at all: that's assumed by a `0` return status.

In general, this convention works really well, and there's no reason "delete user" to say "yep, it's been deleted". That's redundant and unnecessarily verbose. In this case there isn't any useful or filterable output from "delete user" and it probably doesn't matter ... but for consistency it ought to generate no output at all or _at least_ generate that output to `stderr`.

We might look at this as an extension of the old Samuel Clemens advice on editing: whenever you feel inclined to write the word "very", write "damned" instead; your editor will remove it, and the result will be just as it should be". When you feel inclined to write redundant success output, write it to `stderr` -- this will quite correctly feel wrong, and when you remove it entirely, all will be as it should. ;-)

Reply:
Thanks @dbutenhof that was interesting, I am now confused whether to remove the redundant stdout or pipe it to stderr :)

Reply:
Ha; I vote to remove it. You need to explain why the command failed, but not why it succeeded: the `0` return status is enough. (Did you know that the failure of the Roman Empire has been traced to the fact that without the concept of the number "0" they had no way to make their C programs succeed?)